mllib/src/main/scala/org/apache/spark/ml/r/SparkRWrappers.scala

```diff
@@ -17,15 +17,41 @@
 
 package org.apache.spark.ml.api.r
 
+import org.apache.spark.SparkException
 import org.apache.spark.ml.{Pipeline, PipelineModel}
 import org.apache.spark.ml.attribute._
 import org.apache.spark.ml.classification.{LogisticRegression, LogisticRegressionModel}
 import org.apache.spark.ml.clustering.{KMeans, KMeansModel}
 import org.apache.spark.ml.feature.{RFormula, VectorAssembler}
-import org.apache.spark.ml.regression.{LinearRegression, LinearRegressionModel}
+import org.apache.spark.ml.regression._
 import org.apache.spark.sql.DataFrame
 
 private[r] object SparkRWrappers {
+  def fitGLM(
+      value: String,
+      df: DataFrame,
+      family: String,
+      lambda: Double,
+      solver: String): PipelineModel = {
+    if (solver.trim != "irls") throw new SparkException("Currently only support irls")
+
+    val formula = new RFormula().setFormula(value)
+    val regex = "^\\s*(\\w+)\\s*(\\(\\s*link\\s*=\\s*\"(\\w+)\"\\s*\\))?\\s*$".r
```

Comment:
Do not use regex here. Extract the names on R side:

~~~R
> b <- binomial(link = "logit")
> b$family
[1] "binomial"
> b$link
[1] "logit"
~~~